### DISCUSSION TITLE
Grid, dynamic sidebar/navbar width with main 100% width

### DISCUSSION THREAD

**Discussion by frederikhors** [tailwindlabs/tailwindcss#1774] [Help] [answered]
I need a simple main div column 100% full.

Sometimes a sidebar can appear, with dynamic width.

### REPL: https://codesandbox.io/s/tailwind-dynamic-sidebar-kw1ku

In this example you can see I'm not able to change sidebar width (which has every time a new width value) and I'm not able to "stretch" main div to 100% width.

How to do?

```html
<div class="min-h-screen grid grid-cols-12">

  <div class="bg-green-500">
    <div style={`width: ${randomWidth}px !important;`}>
      Sidebar. This has dynamic width.
    </div>
  </div>

  <div class="bg-red-500">
    Main. This has full width.
  </div>

</div>
```

https://stackoverflow.com/posts/61816796

**Comment by robbyrice** [comment, marked as answer]
Instead of using the grid utilities you might want to look into using the flexbox utilities for this layout. It might look something like this.

```
<div class="min-h-screen flex">

  <div style="bg-green-500 {`width: ${randomWidth}px !important;`}">
      Sidebar. This has dynamic width.
  </div>

  <div class="bg-red-500 flex-grow">
    Main. This has full width.
  </div>

</div>
```

**Comment by frederikhors** [comment]
I fixed it with flexbox. Thanks.

**Reply by robbyrice** [reply]
You're welcome. Happy to have helped.
